openlcs/libs/corgi.py

```diff
@@ -345,6 +346,23 @@ def get_product_version(self, name, fields=None) -> dict:
                 retval[field] = product_data[field]
         return retval
 
+    def deduplicate_provides(self, oci_noarch_provides):
+        # Deduplicate provides for components with different arch.
+        nvrs = []
+        provides_rs = []
+        provides_rpm = []
+        for oci_noarch_provide in oci_noarch_provides:
+            purl = oci_noarch_provide.get('purl')
+            purl_dict = PackageURL.from_string(purl).to_dict()
+            if purl_dict.get('namespace') == 'redhat':
+                nvr = get_nvr_from_provide(oci_noarch_provide)
+                if nvr not in nvrs:
+                    nvrs.append(nvr)
+                    provides_rpm.append(oci_noarch_provide)
+            else:
+                provides_rs.append(oci_noarch_provide)
```

Comment:
Then from L354:
```
        for provide in oci_noarch_provides:
            purl = provide.get('purl')
            if purl.startswith('pkg:rpm'):
                nvr = get_nvr_from_purl(purl)
            ...
```

Reply:
Updated.

Reply:
Here doesn't have to use two lists for provides,  one is enough, but it's ok to leave it as it is.

Reply:
Updated :)